core/domain/translation_domain.py

```diff
@@ -18,9 +18,258 @@
+
+    # Private field to track translatable contents in a BaseTranslatableObject.
+    _translatable_contents: Dict[str, TranslatableContent] = {}
+
+    def _register_all_translatable_fields(self) -> None:
+        """Method to register all translatable fields in a translatable object.
+
+        Raises:
+            NotImplementedError. The derived child class must implement the
+                necessary logic to register all translatable fields in an
+                entity.
+        """
+        raise NotImplementedError('Must be implemented in subclasses.')
+
+    def _register_translatable_field(
+        self,
+        field_type: str,
+        content_id: str,
+        value: str|List[str]
+    ) -> None:
+        """Method to register a translatable field in a translatable object.
+
+        Args:
+            field_type: str. The type of the corresponding translatable content.
+            content_id: str. The id of the corresponding translatable content.
+            value: str|list(str). Value of the content which is translatable.
+        """
+        if content_id in self._translatable_contents:
+            raise Exception(
+                'A translatable field is already registered with the '
+                'same content id: %s' % content_id)
+
+        self._translatable_contents[content_id] = TranslatableContent(
+            content_id, value, field_type)
+
+    def _register_translatable_object(
+        self,
+        value: BaseTranslatableObject
+    ) -> None:
+        """Method to register translatable field in a translatable object.
+
+        Args:
+            value: BaseTranslatableObject. An object representing
+                BaseTranslatableObject.
+        """
+        self._translatable_contents.update(value.get_translatable_fields())
+
+    def get_translatable_fields(self) -> Dict[str, TranslatableContent]:
+        """Method to get all the translatable fields in a translatable object.
+
+        Returns:
+            Dict(str, TranslatableContent). Returns the dict containg content_id
+            as key and TranslatableContent as value.
+        """
+        self._translatable_contents = {}
+        self._register_all_translatable_fields()
+        return copy.deepcopy(self._translatable_contents)
+
+    def get_all_contents_which_needs_translations(
+        self,
+        entityTranslation: EntityTranslation
+    ) -> List[TranslatableContent]:
+        """Returns a list of TranslatableContent instances which needs
+        translation or which needs updates in an existing translations.
+
+        Args:
+            entityTranslation: EntityTranslation. An object storing existing
+                translation of an entity.
+
+        Returns:
+            list(TranslatableContent). Returns a list of TranslatableContents.
+        """
+        contents_which_need_translation = []
+        content_ids_for_translated_contents = (
+            entityTranslation.translations.keys())
```

Comment:
Why are we calling this entityTranslation? We only use lowerCamelCase in javascript.

Reply:
Updated.